document-store/src/test/java/org/hypertrace/core/documentstore/model/subdoc/SubDocumentTest.java

```diff
@@ -0,0 +1,55 @@
+package org.hypertrace.core.documentstore.model.subdoc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class SubDocumentTest {
+  @Test
+  void testValidSubDocumentPaths() {
```

Comment:
Thanks for adding the unit test! 
If you could make both the tests parameterized using value source, it would follow the Open-Closed Principle; adding new checks become easier. 

Also, could you add a test to check if underscores are allowed as well.